include/cpr/async.h

```diff
@@ -0,0 +1,48 @@
+#ifndef CPR_ASYNC_H
+#define CPR_ASYNC_H
+
+#include "singleton.h"
+#include "threadpool.h"
+
+namespace cpr {
+
+class GlobalThreadPool : public ThreadPool {
+    SINGLETON_DECL(GlobalThreadPool)
```

Comment:
Any reason for not using something like:
```c++
static GlobalThreadPool& getInstance() {
    static GlobalThreadPool instance;
    return instance;
}
```

Reply:
This is also possible, static object may be not recommended.